sources/osgUtil/Composer.js

```diff
@@ -1146,10 +1160,10 @@ MACROUTILS.createPrototypeObject(
                 '  vec4 rowx = -fac0*texel5 + fac0*texel1 +  -fac1*texel6 + fac1*texel0 + -fac1*texel4 + fac1*texel2;',
                 '  vec4 rowy = -fac0*texel3 + fac0*texel7 +  -fac1*texel4 + fac1*texel6 + -fac1*texel2 + fac1*texel0;',
                 '  float mag = sqrt(dot(rowy,rowy)+dot(rowx,rowx));',
-                '  if (mag < 1.0/255.0) discard;',
-                '  mag *= factor;',
-                '  mag = min(1.0, mag);',
-                '  gl_FragColor = vec4(color*mag,mag);',
+                '  vec3 texelCenter = texture2D(Texture0, vTexCoord0).rgb;',
+                '  if (texelCenter == vec3(0.0) || (uHighlightFactor == 0.0 && mag < 1.0/255.0)) discard;',
```

Comment:
why it's && and not || on this part  (uHighlightFactor == 0.0 && mag < 1.0/255.0)